src/main.cpp

```diff
@@ -115,748 +160,765 @@ int main ( int argc, char** argv )
-                                                .arg ( argv[0] )
-                                                .arg ( ServerOnlyOptions.join ( ", " ) ) );
-            exit ( 1 );
+            // Unknown option ------------------------------------------------------
+            throw CErrorExit ( qUtf8Printable ( QString ( "%1: Unknown option '%2' -- use '--help' for help" ).arg ( argv[0] ).arg ( argv[i] ) ) );
         }
 
-        // mute my own signal in personal mix is only supported for headless mode
-        if ( bUseGUI && bMuteMeInPersonalMix )
-        {
-            bMuteMeInPersonalMix = false;
-            qWarning() << "Mute my own signal in my personal mix is only supported in headless mode.";
-        }
+        // Dependencies ------------------------------------------------------------
 
-        // adjust default port number for client: use different default port than the server since
-        // if the client is started before the server, the server would get a socket bind error
-        if ( !bCustomPortNumberGiven )
+#if ( defined( SERVER_BUNDLE ) && defined( Q_OS_MACX ) ) || defined( SERVER_ONLY )
+        // if we are on MacOS and we are building a server bundle or requested build with serveronly, start Jamulus in server mode
+        if ( bIsClient )
         {
-            iPortNumber += 10; // increment by 10
-            qInfo() << qUtf8Printable ( QString ( "- allocated port number: %1" ).arg ( iPortNumber ) );
+            bIsClient = false;
+            qInfo() << "- Starting in server mode by default (due to compile time option)";
         }
-    }
-    else
-    {
-        if ( ClientOnlyOptions.size() != 0 )
+#endif
+
+#ifdef HEADLESS
+        Q_UNUSED ( bStartMinimized )       // avoid compiler warnings
+        Q_UNUSED ( bShowComplRegConnList ) // avoid compiler warnings
+        Q_UNUSED ( bShowAnalyzerConsole )  // avoid compiler warnings
+        Q_UNUSED ( bMuteStream )           // avoid compiler warnings
+
+        if ( bUseGUI )
         {
-            qCritical() << qUtf8Printable (
-                QString ( "%1: Client only option(s) '%2' used.  See '--help' for help" ).arg ( argv[0] ).arg ( ClientOnlyOptions.join ( ", " ) ) );
-            exit ( 1 );
+            bUseGUI = false;
+            qWarning() << "No GUI support compiled. Running in headless mode.";
         }
+#endif
 
-        if ( bUseGUI )
+#if defined( Q_OS_IOS )
+        bIsClient = true; // Client only - TODO: maybe a switch in interface to change to server?
+        bUseGUI   = true;
+        // bUseMultithreading = true;
+#endif
+
+        // TODO create settings in default state, if loading from file do that next, then come back here to
+        //      override from command line options, then create client or server, letting them do the validation
+        //
+        // See https://github.com/pgScorpio/jamulus/tree/first-big-thing-for-settings for a possible solution on that
```

Comment:
Please remove this comment since it's just a short therm one. I hope in future we'd have a PR.

Reply:
The TODO is not my comment....
And I added the link to my solution in another attempt to get the settings discussion going... (After several failed attempts.)
Nevertheless, there are still a lot PR to merge in main, so I will remove it eventually... ;-)

Reply:
Still the link seems to be new. I'd rather NOT have future fixes in the code. Either fix them right away - that's out of scope of this PR - or document it on GitHub.

Reply:
> Still the link seems to be new. 

Yes. As I said "I added the link"... "in another attempt to get the settings discussion going"

But I get your point... I hope you get my point too. ;=))

Reply:
What ann0see said.